tests/mocha/block_json_test.js

```diff
@@ -65,6 +65,10 @@ suite('Block JSON initialization', function() {
       this.assertError(['test', 2, 'test'], 1,
           'Block "test": Message index %2 out of range.');
     });
+
+    test('Accepts newline token', function() {
```

Comment:
Nit: For new tests I prefer names that follow the format: *tests that* 'my test name here'.

Eg: 'newline characters are accepted', 'newlines are converted to dummy inputs with endOfRow set'. etc.

See here for reasoning: https://testing.googleblog.com/2014/10/testing-on-toilet-writing-descriptive.html